wagtail/wagtailadmin/compare.py

```diff
@@ -0,0 +1,442 @@
+        """
+        Returns True if this child object was created since obj_a
+        """
+        return self.obj_b and not self.obj_a
+
+    def is_deletion(self):
+        """
+        Returns True if this child object was deleted in obj_b
+        """
+        return self.obj_a and not self.obj_b
+
+    def get_position_change(self):
+        """
+        Returns the change in position as an integer. Positive if the object
+        was moved down, negative if it moved up.
+
+        For example: '3' indicates the object moved down three spaces. '-1'
+        indicates the object moved up one space.
+        """
+        if not self.is_addition() and not self.is_deletion():
+            sort_a = getattr(self.obj_a, 'sort_order', 0) or 0
+            sort_b = getattr(self.obj_b, 'sort_order', 0) or 0
+            return sort_b - sort_a
+
+    def get_field_comparisons(self):
+        """
+        Returns a list of comparisons for all the fields in this object.
+        Fields that haven't changed are included as well.
+        """
+        comparisons = []
+
+        if self.is_addition() or self.is_deletion():
+            # Display the fields without diff as one of the versions are missing
+            obj = self.obj_a or self.obj_b
+
+            for field_comparison in self.field_comparisons:
+                comparisons.append(field_comparison(obj, obj))
+        else:
+            for field_comparison in self.field_comparisons:
+                comparisons.append(field_comparison(self.obj_a, self.obj_b))
+
+        return comparisons
+
+    def has_changed(self):
+        for comparison in self.get_field_comparisons():
+            if comparison.has_changed():
+                return True
+
+        return False
+
+    def get_num_differences(self):
+        """
+        Returns the number of fields that differ between the two
+        objects.
+        """
+        num_differences = 0
+
+        for comparison in self.get_field_comparisons():
+            if comparison.has_changed():
+                num_differences += 1
+
+        return num_differences
+
+
+class TextDiff:
+    def __init__(self, changes):
+        self.changes = changes
+
+    def to_html(self, tag='span', addition_class='addition', deletion_class='deletion'):
+        html = ""
+
+        for change_type, value in self.changes:
+            if change_type == 'equal':
+                html += escape(value)
+            elif change_type == 'addition':
+                html += '<{tag} class="{classname}">{value}</{tag}>'.format(
+                    tag=tag,
+                    classname=addition_class,
+                    value=escape(value)
+                )
+            elif change_type == 'deletion':
+                html += '<{tag} class="{classname}">{value}</{tag}>'.format(
+                    tag=tag,
+                    classname=deletion_class,
+                    value=escape(value)
+                )
+
+        return mark_safe(html)
```

Comment:
This would be a good place to make use of Django's [format_html](https://docs.djangoproject.com/en/1.10/ref/utils/#django.utils.html.format_html), so that we don't have to worry about using mark_safe / escape in the right places.